### ISSUE TITLE
IIP+IIQ+IIU

### ISSUE THREAD

**Issue by JerryDerik** [muccc/iridium-toolkit#25]
Good day. Have you figured out a frame merge mechanism to get a full IP, 
аnd how to unite on the counter?

**Comment by Sec42** [comment]
I did some more investigation and got more insights, but it's not yet in a releasable state. Was busy with porting stuff for gr3.8 / python3, so it will take a little longer.

**Comment by JerryDerik** [comment]
Look forward to))). While trying to adapt the system for Windows.

**Comment by muaddib1984** [comment]
I've written a simple script to collect all the IIP messages as hex and ascii, then put them into typical hexdump format. Was hoping to use text2pcap but its obviously not that simple. I'm specifically interested in how to get the session into wireshark. Is it possible to offer a brief explanation of how you manually reassembled the HTTP packets shown in the HOPE talk?

**Comment by JerryDerik** [comment]
So, I transferred the project to Windows. I will not say that this is the easiest solution, but if necessary, I will provide instructions on how to do it.

**Comment by Sec42** [comment]
No, sorry. I didn't spend any time with them in the last years, so there's no new info that I have.

**Comment by muaddib1984** [comment]
Would it be possible to provide the manual process used to reassemble the IIP sessions shown in the original talk? I’ve scratched my head on this for quite a while and can’t seem to re-create your work there.<end transmission>On Feb 25, 2023, at 04:00, Sec ***@***.***> wrote:﻿
No, sorry. I didn't spend any time with them in the last years, so there's no new info that I have.

**Comment by Sec42** [comment]
Honestly, it's been long since i did this, i don't remember enough to describe it without spending more time on this.
I'm leaving this issue open as a reminder to myself to re-visit this, but don't count on it happening soon.
Sorry

**Comment by JerryDerik** [comment]
I can provide algoritm for reassembling IP without IIQ frames.

**Comment by muaddib1984** [comment]
@JerryDerik Sounds good.  Can you provide it somewhere please?

**Comment by JerryDerik** [comment]
Next week i will send link. 20 трав. 2023 р. 22:46 muaddib1984 ***@***.***> пише:
@JerryDerik Sounds good.  Can you provide it somewhere please?

**Comment by dimany4ka1111** [comment]
> Next week i will send link. 20 трав. 2023 р. 22:46 muaddib1984 ***@***.***> пише: @JerryDerik Sounds good. Can you provide it somewhere please? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Hello, can you send me, please, this algorithm???

**Comment by JerryDerik** [comment]
Convert frames in one type of data, found the counter and agregate by it.

**Comment by dimany4ka1111** [comment]
> Convert frames in one type of data, found the counter and agregate by it.

What about IIQ packet types, do you know something?

**Comment by JerryDerik** [comment]
If you cut counter, you could find standart file signatures, like Zip or PDF (if you lucky), and after you can cut it to begin and get information.

**Comment by dimany4ka1111** [comment]
> If you cut counter, you could find standart file signatures, like Zip or PDF (if you lucky), and after you can cut it to begin and get information.

but we don't have a counter in the iiq packet, but I found certain patterns. this is the "cand" field in the header, as well as a different header in the packets (I've only noted two so far), and the most interesting thing is that I saw the transmission of HDLC flags, I even received several IP packets there (I've just gotten the payload for now)

**Comment by JerryDerik** [comment]
![IIQ-CC](https://github.com/user-attachments/assets/ff6b8200-7f54-4076-95f9-fe80290dfa8b)
Look. On IIQ-CC you can see that what you call HDLC. But it`s not. It`s like Control Chanel with "Padding with zeros". Information transmitting in fields with counter in first two bytes minus three first bits.
![IIQ](https://github.com/user-attachments/assets/78a9f972-fb15-4b58-87b0-ac8e8511ca4c)

**Comment by dimany4ka1111** [comment]
Ok, but how you get this IIQ-CC, I don’t have this image, I just get all IIQ packets and get data, but you have another image
Thx for help!

**Comment by JerryDerik** [comment]
You need something like this editor
https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://github.com/WerWolv/ImHex&ved=2ahUKEwjj_NmNuuaJAxXEgSoKHZeZBoQQFnoECB0QAQ&usg=AOvVaw1tMGmjO3Y8xOSafQkDkzio

Then you must build bits on period 248 (if i right remember). Theh you cut first two bytes, but must manualy separate control chanel from information (counter in first two bytes help you).

P. S. Just intrest for my self, wher are you from?)
If dont want to answer just ignore this)

**Comment by dimany4ka1111** [comment]
I am from Latvia

**Comment by dimany4ka1111** [comment]
![IMG_0361](https://github.com/user-attachments/assets/75840a10-f5c3-463a-92ff-95be86c09a8f)
Thx, but I have mine bits editor 
So there is, that I get
After getting all hex raw packets in one packet and converts into bits on the period 248 bits
I think, this is wrong😂

**Comment by JerryDerik** [comment]
> And you, if it isn’t secret) 

Ukraine.

**Comment by JerryDerik** [comment]
> Yes, it is 

So use autoperiod, or manualy search the period from 200+

**Comment by JerryDerik** [comment]
Here my Signal, to not spam here.
https://signal.me/#eu/Hv9XcoJwug8J2baxwLHcT_IwHB1UDGLGQLtezAG-C5JIdphdJS1Kdrrrh-y-_zfb
